core/src/test/scala/integration/kafka/network/DynamicConnectionQuotaTest.scala

```diff
@@ -275,7 +276,7 @@ class DynamicConnectionQuotaTest extends BaseRequestTest {
     val initialConnectionCount = connectionCount
     val entity = new ClientQuotaEntity(Map(ClientQuotaEntity.IP -> ip.orNull).asJava)
     val request = Map(entity -> Map(QuotaConfigs.IP_CONNECTION_RATE_OVERRIDE_CONFIG -> Some(updatedRate.toDouble)))
-    alterClientQuotas(admin, request).all.get()
+    alterClientQuotas(admin, request).all.get(JTestUtils.DEFAULT_MAX_WAIT_MS, TimeUnit.MILLISECONDS)
```

Comment:
Pardon me, why this can fix the issue? I don't catch the root cause actually :_

Reply:
I was thinking altering of quotas is timing out, and before that's finished completely, connectionRateForIp has started to process without enough data, and times out probably ?